lib/WebService/Async/SmartyStreets.pm

```diff
@@ -0,0 +1,199 @@
+package WebService::Async::SmartyStreets;
+# ABSTRACT: Access SmartyStreet API
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=HEAD
+
+WebService::Async::SmartyStreets - calls the SmartyStreets API and checks for the validity of the address
+
+=head1 SYNOPSIS
+    
+    my $ss;
+
+    $ss = WebService::Async::SmartyStreets->new(
+        auth_id => #insert auth_id,
+        token   => #insert token,
+        );
+    IO::Async::Loop->new->add($ss);
+
+    $ss = get_smartystreets();
+
+    my $addr = $ss->verify_international(<hash of address element>, geocode => 'true')->get;
+    print($addr->status);
+    
+=head1 DESCRIPTION
+
+This class calls the SmartyStreets API and parse the response to L<WebService::Async::SmartyStreets::Address>
+
+Note that this module uses L<Future::AsyncAwait>
+
+=over 4
+
+=cut
+
+use parent qw(IO::Async::Notifier);
+
+use mro;
+no indirect;
+
+use URI;
+use URI::QueryParam;
+
+use Future::AsyncAwait;
+use Net::Async::HTTP;
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::SmartyStreets::Address;
+
+use Log::Any qw($log);
+
+=head2 configure
+
+Configures the class with the auth_id and token
+Takes in: Hash which consists of auth_id and token
+
+=cut
+
+sub configure {
+    my ($self, %args) = @_;
+    for my $k (qw(auth_id token)) {
+        $self->{$k} = delete $args{$k} if exists $args{$k};
+    }
+    $self->next::method(%args);
+}
+
+sub auth_id { shift->{auth_id} }
+sub token   { shift->{token} }
+
+sub next_id {
+    ++(shift->{id} //= 'AA00000000');
+}
+
+=head2 ua
+
+Accessor for the L<Net::Async::HTTP> instance which will be used for SmartyStreets API requests.
+
+=cut
+
+sub ua {
+    my ($self) = @_;
+    $self->{ua} //= do {
+        $self->add_child(
+            my $ua = Net::Async::HTTP->new(
+                fail_on_error            => 1,
+                decode_content           => 1,
+                pipeline                 => 0,
+                max_connections_per_host => 4,
+                user_agent =>
+                    'Mozilla/4.0 (WebService::Async::SmartyStreets; BINARY@cpan.org; https://metacpan.org/pod/WebService::Async::SmartyStreets)',
+            ));
+        $ua;
+        }
+}
+
+=head2 verify
+
+Makes connection to SmartyStreets API and parses the response into WebService::Async::SmartyStreets::Address.
+
+    my $ss = WebService::Async::SmartyStreets::International->new(
+        auth_id => "...",
+        token   => "...",
+    );
+    IO::Async::Loop->new->add($ss);
+
+    my $addr = $ss->verify("URI String", %address_to_check)->get;
+
+Takes the following named parameters:
+
+=over 4
+
+=item * C<uri> - URI address (in string)
```

Comment:
POD is a paragraph-based system, all `=directives` need to have a blank line before + after. See http://perldoc.pl/perlpod for more details.